Modules/_elementtree.c

```diff
@@ -908,13 +916,20 @@ deepcopy(elementtreestate *st, PyObject *object, PyObject *memo)
                     break;
                 }
             }
-            if (simple)
+            if (simple) {
+                // This does not call object.__copy__(), so it's still safe.
                 return PyDict_Copy(object);
+            }
             /* Fall through to general case */
         }
         else if (Element_CheckExact(st, object)) {
-            return _elementtree_Element___deepcopy___impl(
-                (ElementObject *)object, memo);
+            // The __deepcopy__() call may call arbitrary code even if the
+            // object to copy is a built-in XML element (one of its children
+            // may mutate 'object' in its own __deepcopy__() implementation).
```

Comment:
Not just "mutate 'object'", but mutated the parent Element, making `object` the hanging pointer.